### PR TITLE
Fix Sarif output driver issues and Xray Sca locations

### PR DESCRIPTION
**Pull request by attiasas**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

Fixes some discovered issues that make the Sarif output not pass [validations](https://sarifweb.azurewebsites.net/Validation):
* Fix the returned path at Xray SCA results to show the actual full path or not at all.
* Add Information Uri for missing Advance security scanners
* Set default version for Advance scanners if it does not exist or is not valid to the AnalyzerManager version.
* Move test funcs from sarifutils to a separate file

* Fix bug in setting the file names of locations
* Remove 'markdown' flag and make the Xray run generation fit for all users

### CODE REVIEW

File: utils/coreutils/techutils.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ type TechData struct {
 	// Whether Contextual Analysis supported in this technology.
 	applicabilityScannable bool
 	// The file that handles the project's dependencies.
-	packageDescriptor string
+	packageDescriptor []string
```
Comment (sverdlov93) [inline utils/coreutils/techutils.go:48]:
```suggestion
	packageDescriptors []string
```

File: utils/coreutils/techutils.go [hunk 2]
```diff
@@ -45,7 +45,7 @@ type TechData struct {
 	// Whether Contextual Analysis supported in this technology.
 	applicabilityScannable bool
 	// The file that handles the project's dependencies.
```
Comment (sverdlov93) [inline utils/coreutils/techutils.go:47]:
```suggestion
	// The files that handle the project's dependencies.
```

File: xray/utils/resultwriter.go [hunk 3]
```diff
@@ -287,6 +288,54 @@ func addXrayCveIssueToSarifRun(cves []formats.CveRow, issueId, severity, file st
 	return nil
 }
 
+func getDescriptorFullPath(tech coreutils.Technology, run *sarif.Run) (string, error) {
+	descriptors := tech.GetPackageDescriptor()
+	if len(descriptors) == 1 {
+		// Generate the full path
+		return GetFullLocationFileName(strings.TrimSpace(descriptors[0]), run.Invocations), nil
+	}
+	for _, optional := range descriptors {
```
Comment (sverdlov93) [inline xray/utils/resultwriter.go:297]:
```suggestion
	for _, descriptor := range descriptors {
```

File: xray/utils/resultwriter.go [hunk 4]
```diff
@@ -287,6 +288,54 @@ func addXrayCveIssueToSarifRun(cves []formats.CveRow, issueId, severity, file st
...
+		// If multiple options return first to match
+		full := GetFullLocationFileName(strings.TrimSpace(optional), run.Invocations)
```
Comment (sverdlov93) [inline xray/utils/resultwriter.go:299]: why not filepath.abs() ?

Comment (sverdlov93) [inline xray/utils/resultwriter.go:299]:
```suggestion
		absolutePath := GetFullLocationFileName(strings.TrimSpace(optional), run.Invocations)
```

File: xray/utils/resultwriter.go [hunk 5]
```diff
@@ -287,6 +288,54 @@ func addXrayCveIssueToSarifRun(cves []formats.CveRow, issueId, severity, file st
...
+	}
+	if strings.TrimSpace(descriptorPath) == "" {
+		// Can't calculate actual file location
+		if markdown {
+			return sarif.NewLocation().WithPhysicalLocation(sarif.NewPhysicalLocation().WithArtifactLocation(sarif.NewArtifactLocation().WithUri(tech.ToFormal() + " Package Descriptor"))), nil
+		}
+		return
+	}
+	return sarif.NewLocation().WithPhysicalLocation(sarif.NewPhysicalLocation().WithArtifactLocation(sarif.NewArtifactLocation().WithUri("file://" + descriptorPath))), nil
+}
```
Comment (sverdlov93) [inline xray/utils/resultwriter.go:337]: ?

File: xray/utils/resultwriter.go (hunk 4 above)
Comment (attiasas) [reply xray/utils/resultwriter.go:299]: this is string, not file

**Pull request merged**